pkg/pillar/base/logobjecttypes.go

```diff
@@ -0,0 +1,291 @@
+// Copyright (c) 2020 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package base
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/satori/go.uuid"
+	log "github.com/sirupsen/logrus"
+)
+
+// LogObjectType : Predefined log object types
+type LogObjectType string
+
+// The following is NOT an extensive list.
+// Intent is to create a reference list for engineers to get an idea of its usage.
+const (
+	UnknownLogType         LogObjectType = ""
+	LogType                LogObjectType = "log"
+	ImageLogType           LogObjectType = "image"
+	RelationLogType        LogObjectType = "relation"
+	AppInstanceLogType     LogObjectType = "app_instance"
+	NetworkInstanceLogType LogObjectType = "network_instance"
```

Comment:
There is a question in my mind whether we should derive these names from the TypeToName function i.e. they will be "AppInstanceStatus".
Part of the question is whether we want log items related to AppInstanceConfig and AppInstanceStatus to be viewed as one or two different log object types; might make sense for them to be one.

We can explore putting pubsub hooks in the subscribe side as well; that way in the example code we will get entries for AppInstanceConfig, which will have different fields (e.g., Activate and RestartCounter/PurgeCounter would be key ones) than the Status.

Reply:
My opinion that we should leave some flexibility with naming the object type. I agree one shouldn't start creating too many arbitrary strings, but that can be discouraged with the usage of constants.

And I have added the hooks to subscriptions part also.
Will do the document also.